File: Data Structures & Algorithms/plus-one/submission-4.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        while True:
            if digits[i] != 9:
                digits[i] += 1
                return digits
            digits[i] = 0
            if i != 0:
                i -= 1
            else:
                digits.insert(0, 1)
                return digits

File: Data Structures & Algorithms/missing-number/submission-4.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        set<int> hashSet = {};
        for (int num : nums) hashSet.insert(num);
        for (int i = 0; i < nums.size() + 1; ++i) {
            if (hashSet.contains(i) == false) return i;
        }
}
};
